src/ThreeEditor/js/Editor.js

```diff
@@ -179,7 +183,21 @@ Editor.prototype = {
 
 	},
 
-	//
+	// YAPTIDE ZONES MAANGER
+	setZonesManager: function(zonesManager) {
+		this.zonesManager.uuid = zonesManager.uuid;
+		this.zonesManager.zonesContainer.uuid = zonesManager.zonesContainer.uuid;
+		// avoid render per object
+
+		this.signals.sceneGraphChanged.active = false;
+
+		while (zonesManager.zonesContainer.children.length > 0) {
```

Comment:
Is it guaranteed that this loop will end in finite time ?

Reply:
I made it exactly the same way the ThreeJS editor handles loading from JSON regular scenes. 
Basically, Editor.addObject() removes the object from its parent's children list making sure this loop will have finite execution. CSGManager.addZone() only applies worker to added zone before executing same function.

Reply:
Good, thanks for explanation !